src/should-forward-property.js

```diff
@@ -600,17 +26,18 @@ const isCustomAttribute = RegExp.prototype.test.bind(
 )
 
 const hasItem = (list, name) => list.indexOf(name) !== -1
-const isHtmlProp = name => hasItem(htmlProps, name)
+const isHtmlProp = (name, tagName) => {
+  const elementAttributes = reactHTMLAttributes[tagName]
+
+  return hasItem([...globalReactHtmlProps, ...elementAttributes], name)
```

Comment:
> Another idea is caching the result of shouldForwardProperty in its file scope by using closure so that the combination of tagName and name that has already been checked before doesn't need to go through the same process again to get the result.

Oh yeah, and memoization is what you're talking about and that would probably be useful! I don't know if there's a super small memoization library, but might be worth looking into before implementing it ourselves :)